### PR TITLE
Add a reporter Id

### PR DESCRIPTION
**Pull request by SlimaneAmar**
Signed-off-by: Slimane AMAR <slimane.amar@rte-france.com>

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 1]
```diff
@@ -218,13 +220,20 @@ private Mono<SecurityAnalysisResult> run(SecurityAnalysisRunContext context, UUI
         return Mono.zip(network, contingencies)
                 .flatMap(tuple -> {
 
-                    Reporter reporter = context.getReportUuid() != null ? new ReporterModel("SecurityAnalysis", "Security analysis") : Reporter.NO_OP;
+                    Reporter rootReporter = Reporter.NO_OP;
+                    Reporter reporter = Reporter.NO_OP;
+                    if (context.getReportUuid() != null) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java:225]: context.getReporterId() != null ?

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 2]
```diff
@@ -218,13 +220,20 @@ private Mono<SecurityAnalysisResult> run(SecurityAnalysisRunContext context, UUI
...
+                        String rootReporterId = context.getReporterId() + "@" + AS_TYPE_REPORT;
+                        rootReporter = new ReporterModel(rootReporterId, rootReporterId);
+                        reporter = rootReporter.createSubReporter(AS_TYPE_REPORT, AS_TYPE_REPORT + " (${providerToUse})", "providerToUse", context.getProvider());
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java:228]: Maybe check if context.getProvider() is null ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java:225]: Done

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java]: Done

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**